src/NLU.DevOps.ModelPerformance/NLUAccuracy.cs

```diff
@@ -0,0 +1,116 @@
+        }
+
+        /// <summary>
+        /// Calculates the f1 score from a confusion matrix
+        /// </summary>
+        /// <param name="cm"> confusion matrix metrics</param>
+        /// <returns> the f1 result</returns>
+        public static double F1(ConfusionMatrix cm)
+        {
+            var precision = cm.Precision();
+            var recall = cm.Recall();
+            var denominator = precision + recall;
+            return denominator != 0 ? 2 * (precision * recall) / denominator : 0;
+        }
+
+        /// <summary>
+        /// Calculates Precision, Recall and F1 Score
+        /// </summary>
+        /// <param name="cm">confusion matrix metrics</param>
+        /// <returns>List that contains the results calculated</returns>
+        public static List<double> CalcMetrics(ConfusionMatrix cm)
+        {
+            List<double> metrics = new List<double>
+            {
+                Precision(cm),
+                Recall(cm),
+                F1(cm)
+            };
+            return metrics;
+        }
+
+        /// <summary>
+        /// Prints to the console the intents and entities performance results in a table
+        /// </summary>
+        /// <param name="statistics"> The computed data for intents and entities</param>
+        public static void PrintResults(this NLUStatistics statistics)
+        {
+            const int roundingplace = 4;
+
+            Console.WriteLine("== Intents results == ");
+            var intentTable = new ConsoleTable("Intent", "Precision", "Recall", "F1");
+            var intentsTotalResults = NLUAccuracy.CalcMetrics(statistics.Intent).Select(intent => Math.Round(intent, roundingplace)).ToList();
+            intentTable.AddRow("*", intentsTotalResults[0], intentsTotalResults[1], intentsTotalResults[1]);
+
+            foreach (KeyValuePair<string, ConfusionMatrix> kvp in statistics.ByIntent)
+            {
+                // Calculating accuracy and rounding up the result values
+                var results = NLUAccuracy.CalcMetrics(kvp.Value).Select(value => value = Math.Round(value, roundingplace)).ToList();
+                intentTable.AddRow(kvp.Key, results[0], results[1], results[2]);
+            }
+
+            intentTable.Write();
+            Console.WriteLine();
+
+            Console.WriteLine("== Entity results == ");
+            var entityTable = new ConsoleTable("Entity", "Precision", "Recall", "F1");
+            var entityTotalResults = NLUAccuracy.CalcMetrics(statistics.Entity).Select(value => value = Math.Round(value, roundingplace)).ToList();
+            entityTable.AddRow("*", entityTotalResults[0], entityTotalResults[1], entityTotalResults[2]);
+
+            foreach (KeyValuePair<string, ConfusionMatrix> kvp in statistics.ByEntityType)
+            {
+                var entityResults = NLUAccuracy.CalcMetrics(kvp.Value).Select(value => value = Math.Round(value, roundingplace)).ToList();
```

Comment:
No reason to calculate these separately, might actually be easier to read if you just did:
```
var cm = kvp.Value;
entityTable.AddRow(kvp.Key, cm.Precision(), cm.Recall(), cm.F1());
```

Reply:
hmm yeah makes sense. The reason I did it this way it to do the decimal rounding in one shot instead of polluting the AddRow line.

Reply:
I actually ended up keeping it but simplifying the logic for choosing the number of decimals.